nodebuilder/header/service.go

```diff
@@ -51,6 +52,33 @@ func (s *Service) GetVerifiedRangeByHeight(
 }
 
 func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	head, err := s.syncer.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(head.Height()+1) < height {
```

Comment:
should just be `head.Height()` here, not +1 because even if we allow for some lag, it still might be inaccessible for some time due to sync process.

Reply:
The sync time for one header is negligible. The reason I want this because I believe this is great UX enhancer that prevents our user to see failures for events that are millisecond away.

Reply:
Ok.